doc: Add a comment describing the algorithm used by entropy_nrf5.c

SebastianBoe opened the pull request:
Hi,

I made some clarifying notes to make sense of the entropy driver. If the notes are correct,
I think it will make it easier for users to understand the rationale behind the algorithm and
how to configure it for maximum performance in their designs.

Please correct if I have misunderstood anything.

Signed-off-by: Sebastian Bøe <sebastian.boe@nordicsemi.no>

carlescufi commented on drivers/entropy/entropy_nrf5.c:
```diff
@@ -19,8 +72,8 @@ struct rand {
 
-#define RAND_THREAD_LEN (CONFIG_ENTROPY_NRF5_THR_BUF_LEN + 1)
-#define RAND_ISR_LEN    (CONFIG_ENTROPY_NRF5_ISR_BUF_LEN + 1)
+#define RAND_THREAD_LEN (CONFIG_ENTROPY_NRF5_THR_THRESHOLD + 1)
```
This defines the total length of the buffer, which is BUF_LEN. The THRESHOLD is only there to trigger the hardware.

SebastianBoe replied:
Oops, I am really sorry, this patch was not supposed to be included.

SebastianBoe committed:
doc: Add a comment describing the algorithm used by entropy_nrf5.c

Improve the documentation of entropy_nrf5.c by adding a comment that
clarifies it's characteristics.

Signed-off-by: Sebastian Bøe <sebastian.boe@nordicsemi.no>
